apps/ledger-live-mobile/src/components/RequiresBLE/hooks/useAndroidLocationPermission.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  Permission,
+  PermissionsAndroid,
+  Platform,
+  AppState,
+} from "react-native";
+import { useRef, useCallback, useState, useEffect } from "react";
+
+const { PERMISSIONS, RESULTS } = PermissionsAndroid;
+
+// https://developer.android.com/guide/topics/connectivity/bluetooth/permissions#declare
+export const locationPermission: Permission | undefined =
+  Platform.OS === "android"
+    ? Platform.Version <= 28
+      ? PERMISSIONS.ACCESS_COARSE_LOCATION
+      : PERMISSIONS.ACCESS_FINE_LOCATION
+    : undefined;
+
+async function checkLocationPermission() {
+  if (!locationPermission) return true;
+  return PermissionsAndroid.check(locationPermission);
+}
+
+type RequestResult = {
+  granted: boolean;
+  status: typeof RESULTS[number];
+};
+
+async function requestLocationPermission(): Promise<RequestResult> {
+  if (!locationPermission) return { granted: true, status: RESULTS.GRANTED };
+  return PermissionsAndroid.request(locationPermission).then(status => {
+    return {
+      granted: status === RESULTS.GRANTED,
+      status,
+    };
+  });
+}
+
+const useAndroidLocationPermission = () => {
+  const [hasPermission, setHasPermission] = useState<boolean | undefined>();
+  const [check, setCheck] = useState(false);
+  const [neverAskAgain, setNeverAskAgain] = useState(false);
+  const [retryNonce, setRetryNonce] = useState(0);
+  const timeout = useRef<ReturnType<typeof setTimeout> | null>(null);
+
+  const checkAgain = useCallback(() => {
+    setRetryNonce(new Date().getTime());
+  }, []);
+
+  useEffect(() => {
+    // set hasPermission to false after 400ms if it's still undefined.
+    // without this we risk a black screen when this unmounts or a blink
+    // if we ignore it. Feel free to come up with a better way.
+    timeout.current = setTimeout(() => {
+      setHasPermission(false);
+    }, 200);
+  }, []);
```

Comment:
Same: should we clear the timeout in the useEffect-cleanup-return-function just in case though ?